core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -84,6 +84,11 @@ private AllDataFilesTableScan(TableOperations ops, Table table, Schema schema, S
       this.fileSchema = fileSchema;
     }
 
+    @Override
+    protected String tableType() {
+      return String.valueOf(MetadataTableType.ALL_DATA_FILES);
```

Comment:
I think these can all be .name() to get the value we are looking for

Reply:
That's a great call. The `String.valueOf` was making me pretty sick tbh.